Site/SiteAnalyticsModule.php

```diff
@@ -389,6 +584,186 @@ public function getFacebookPixelCommandsInlineJavascript()
+
+	protected function displayFacebookPixel()
+	{
+		if ($this->hasFacebookPixel()) {
+			$js = $this->getFacebookPixelInlineJavascript();
+			if ($js != '') {
+				Swat::displayInlineJavaScript($js);
+			}
+		}
+	}
+
+	// }}}
+	// {{{ protected function getFacebookPixelCommand()
+
+	protected function getFacebookPixelCommand($command)
+	{
+		if (!is_array($command)) {
+			$command = array($command);
+		}
+
+		return sprintf(
+			'fbq(%s);',
+			implode(', ', array_map('json_encode', $command))
+		);
+	}
+
+	// }}}
+
+	// Twitter
+	// {{{ public function hasTwitterPixel()
+
+	public function hasTwitterPixel()
+	{
+		return (
+			$this->twitter_track_pixel_id != '' &&
+			!$this->analytics_opt_out
+		);
+	}
+
+	// }}}
+	// {{{ public function pushTwitterPixelCommands()
+
+	public function pushTwitterPixelCommands(array $commands)
+	{
+		foreach ($commands as $name => $value) {
+			$this->twitter_pixel_commands[$name] = $value;
+		}
+	}
+
+	// }}}
+	// {{{ public function prependTwitterPixelCommands()
+
+	public function prependTwitterPixelCommands(array $commands)
+	{
+		$comands = array_reverse($commands);
+		foreach ($commands as $command) {
+			array_unshift($this->twitter_pixel_commands, $command);
+		}
+	}
+
+	// }}}
+	// {{{ public function getTwitterPixelImages()
+
+	public function getTwitterPixelImages()
+	{
+		$xhtml = <<<'XHTML'
+<noscript>
+<img height="1" width="1" style="display:none;" alt="" src="https://analytics.twitter.com/i/adsct?txn_id=%1$s&p_id=Twitter" />
+<img height="1" width="1" style="display:none;" alt="" src="//t.co/i/adsct?txn_id=%1$s&p_id=Twitter" />
+<img height="1" width="1" style="display:none;" alt="" src="https://analytics.twitter.com/i/adsct?txn_id=%2$s&p_id=Twitter&%3$s" />
+<img height="1" width="1" style="display:none;" alt="" src="//t.co/i/adsct?txn_id=%2$s&p_id=Twitter&%3$s" />
+</noscript>
+XHTML;
+
+		$track_pixel = rawurlencode($this->twitter_track_pixel_id);
+		$purchase_pixel = rawurlencode($this->twitter_purchase_pixel_id);
+
+		$query_vars = array();
+		foreach ($this->twitter_pixel_commands as $name => $value) {
+			$query_vars[$name] = sprintf(
+				'%s=%s',
+				SwatString::minimizeEntities(rawurlencode($name)),
+				SwatString::minimizeEntities(rawurlencode($value))
+			);
+		}
+
+		return sprintf(
+			$xhtml,
+			SwatString::minimizeEntities($track_pixel),
+			SwatString::minimizeEntities($purchase_pixel),
+			implode('&', $query_vars)
+		);
+	}
+
+	// }}}
+	// {{{ public function getTwitterPixelInlineJavascript()
+
+	public function getTwitterPixelInlineJavascript()
+	{
+		$javascript = null;
+
+		if ($this->hasTwitterPixel()) {
+			$javascript = $this->getTwitterPixelTrackerInlineJavascript();
+		}
+
+		return $javascript;
+	}
+
+	// }}}
+	// {{{ public function getTwitterPixelTrackerInlineJavascript()
+
+	public function getTwitterPixelTrackerInlineJavascript()
+	{
+		$javascript = <<<'JS'
+(function() {
+var ga = document.createElement('script');
```

Comment:
Is this code cribbed from twitter? The `ga` variable name is confusing. Suggest maybe using `tp` instead of `ga`.